api/envoy/type/percent.proto

```diff
@@ -46,3 +46,18 @@ message FractionalPercent {
   // fractional percentage is capped at 1 (100%).
   DenominatorType denominator = 2 [(validate.rules).enum.defined_only = true];
 }
+
+// Runtime derived FractionalPercent with defaults for when the numerator or denominator is not
+// specified via a runtime key.
+message RuntimeFractionalPercent {
+  // Default value if the runtime value's for the numerator/denominator keys are not available.
+  FractionalPercent default_value = 1 [(validate.rules).message.required = true];
+
+  // Runtime key to get numerator value for comparison. This value is used if defined.
+  string numerator_runtime_key = 2;
```

Comment:
i think you could safely shorten these to numerator_key, denominator_key since they're already under runtime_fraction. up to you.